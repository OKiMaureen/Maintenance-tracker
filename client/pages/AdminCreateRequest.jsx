import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRequestAction from '../actions/createRequestAction';
import Header from '../components/Header';
import clearMessageAction from '../actions/clearMeassageAction';
import logoutAction from '../actions/logoutAction';
import Request from '../components/Request';

export class CreateRequest extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      department: '',
      equipment: '',
      serialnumber: '11111111',
      title: '',
      description: '',
      errors: {},
    }
    validate = () => {
      const validations = new Validator(
        {
          department: this.state.department,
          equipment: this.state.equipment,
          serialnumber: this.state.serialnumber,
          title: this.state.title,
          description: this.state.description,
        },

        {
          title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
          department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
          description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
        },
        {
          'min.title': 'The :attribute must not be less than 5 characters.',
          'max.title': 'The :attribute must not be greater than 20 characters.',
          'min.serialnumber': 'The :attribute must be only 8 characters.',
          'max.serialnumber': 'The :attribute must be only 8 characters.',
          'min.description': 'The :attribute must not be less than 3 characters.',
          'max.description': 'The :attribute must not be greater than 50 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      const { clearMessage, requestDetail } = this.props;
      if (requestDetail.error) clearMessage();
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.createRequestAction(this.state, this.props.history);
      }
      return true;
    }
    logout = () => {
      const { history, logout } = this.props;
      logout();
      localStorage.clear();
      history.push('/');
      return true;
    }
    render() {
      const { requestDetail, userDetail } = this.props;
      const {
        title, description, equipment, errors, department,
      } = this.state;
      return (
        <div>
          <Header><Link to="/">Home</Link><Link to="/admincreaterequest">Create request</Link> <Link to="/adminrequests">All requests</Link><li><button className="header-btn" onClick={this.logout}>Logout</button></li></Header>
          <Request
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            title={title}
            description={description}
            equipment={equipment}
            requestDetail={requestDetail}
            userDetail={userDetail}
            errors={errors}
            department={department}
            titleText="Create Request"
            buttonText="Create Request"
          />
        </div>
      );
    }
}

const mapStateToProps = state => ({
  requestDetail: state.createRequest,
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  createRequestAction,
  clearMessage: clearMessageAction,
  logout: logoutAction,
}, dispatch);

CreateRequest.propTypes = {
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  createRequestAction: PropTypes.func.isRequired,
  clearMessage: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(CreateRequest);

