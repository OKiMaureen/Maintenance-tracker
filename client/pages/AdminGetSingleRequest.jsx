import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import i from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import adminGetSingleRequestAction, { updateReq, disapproveRequest, resolveRequest } from '../actions/adminSingleRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AdminSingleRequest extends Component {
  componentDidMount() {
    const { adminSingleRequest, match } = this.props;
    adminSingleRequest(match.params.id);
  }
  logout = () => {
    const { history, logout } = this.props;
    logout();
    localStorage.clear();
    history.push('/');
    return true;
  }
  updateRequest = (e) => {
    const status = e.target.id;
    const { update, oneRequest } = this.props;
    update(oneRequest.id, status);
    return true;
  }

  render() {
    const {
      oneRequest,
    } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/adminrequests">All requests</Link>
          <li><button className="header-btn" onClick={this.logout}>Logout</button></li>
        </Header>

        <section className="allrequests">
          {oneRequest ?
            <div className="detailsrequest-card" key={i.generate()}>
              <div className="status">
                {oneRequest.requeststatus === 'approved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="approve" className="disabled" disabled onClick={this.updateRequest}>Approve</button> :
                  <button className="green" id="approve" onClick={this.updateRequest}>Approve</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="disapprove" className="disabled" disabled onClick={this.updateRequest}>Disapprove</button> :
                  <button className="red" id="disapprove" onClick={this.updateRequest}>Disapprove</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' || oneRequest.requeststatus === 'pending' ?
                  <button id="resolved" className="disabled" disabled onClick={this.updateRequest}>Resolve</button> :
                  <button className="green" id="resolve" onClick={this.updateRequest}>Resolve</button>}
              </div>
              <p>
                <label>Title:</label>{oneRequest.title}
              </p>
              <p>
                <label>Department:</label>{oneRequest.department}
              </p>
              <p>
                <label>Equipment: </label>{oneRequest.equipment}
              </p>
              <p>
                <label>Description: </label>{oneRequest.description}
              </p>
              <div className="container">
                <div className="status fixed">
                  {oneRequest.requeststatus === 'pending' ? <label className="yellow">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'approved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'disapproved' ? <label className="red">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'resolved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                </div>
              </div>
            </div> : '' }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  oneRequest: state.adminSingleRequest.request,
  userDetail: state.authUser,
  logout: logoutAction,

});
const mapDispatchToProps = dispatch => bindActionCreators({
  adminSingleRequest: adminGetSingleRequestAction,
  update: updateReq,
  disapprove: disapproveRequest,
  resolve: resolveRequest,

}, dispatch);

AdminSingleRequest.propTypes = {
  oneRequest: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  adminSingleRequest: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AdminSingleRequest);
