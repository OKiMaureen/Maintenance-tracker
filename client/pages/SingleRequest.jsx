import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import i from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import getARequestAction from '../actions/singleRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class SingleRequest extends Component {
  componentDidMount() {
    const { aRequest, match } = this.props;
    aRequest(match.params.id);
  }
  logout = () => {
    const { history, logout } = this.props;
    logout();
    localStorage.clear();
    history.push('/');
    return true;
  }
  render() {
    const { oneRequest } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/allrequests">My request</Link>
          <li><button className="header-btn" onClick={this.logout}>Logout</button></li>
        </Header>

        <section className="allrequests">


          {oneRequest ?
            <div className="detailsrequest-card" key={i.generate()}>
              <p>
                <label>Title:</label>{oneRequest.title}
              </p>
              <p>
                <label>Department:</label>{oneRequest.department}
              </p>
              <p>
                <label>Equipment: </label>{oneRequest.equipment}
              </p>
              <p>
                <label>Description: </label>{oneRequest.description}
              </p>
              <div className="container">
                <div className="status fixed">
                  {oneRequest.requeststatus === 'pending' ? <label className="yellow">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'approved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'disapproved' ? <label className="red">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'resolved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                </div>
                <div className="status fixed ">
                  <p>
                    {oneRequest.requeststatus === 'pending' ? <Link to={`/editrequest/${oneRequest.id}`}>Edit</Link> : ''}

                  </p>
                </div>
              </div>
            </div> :
            <div className="requests-card no-request" >
              <p className="message-centered">
              You do not have any requests yet!!! Go ahead and create a request.
              </p>
            </div>
          }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  oneRequest: state.singleRequest.request,
  userDetail: state.authUser,
  logout: logoutAction,

});
const mapDispatchToProps = dispatch => bindActionCreators({
  aRequest: getARequestAction,
}, dispatch);

SingleRequest.propTypes = {
  oneRequest: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  aRequest: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(SingleRequest);
