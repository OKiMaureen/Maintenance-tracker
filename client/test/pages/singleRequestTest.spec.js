import React from 'react';
import { shallow } from 'enzyme';
import { SingleRequest } from '../../pages/SingleRequest';


describe('<SingleRequest />', () => {
  let wrapper;
  const match = {
    params: {
      id: '1',
    },
  };
  let props = {
    aRequest: () => {},
    logout: () => {},
    oneRequest:
      {
        id: 1,
        user_id: 1,
        title: 'repair computer',
        department: 'technology',
        equipment: 'computer',
        requeststatus: 'pending',
        description: 'repair laptop',
        serialnumber: 111111111,
      },
    history: {
      push: jest.fn(),
    },

  };
  it('Should display request card when request loads and it exists', () => {
    props = {
      aRequest: () => {},
      oneRequest: {
        id: 1,
        user_id: 1,
        title: 'repair computer',
        department: 'technology',
        equipment: 'computer',
        requeststatus: 'pending',
        description: 'repair laptop',
        serialnumber: 111111111,
      },
    };

    wrapper = shallow(<SingleRequest {...props} match={match} />);
    expect(wrapper.find('label').exists()).toBe(true);
  });
  it('Should handle logout method', () => {
    props = {
      aRequest: () => {},
      logout: () => {},
      history: {
        push: jest.fn(),
      },
    };
    wrapper = shallow(<SingleRequest {...props} match={match} />);
    expect(wrapper.instance().logout()).toBe(true);
  });
  it('Should display request label as yellow when status is pending', () => {
    props = {
      aRequest: () => {},
      oneRequest:
        { requeststatus: 'pending' },
    };

    wrapper = shallow(<SingleRequest {...props} match={match} />);
    expect(wrapper.find('.yellow').exists()).toBe(true);
  });
  it('Should display request label as green when status is approve', () => {
    props = {
      aRequest: () => {},
      oneRequest:
        { requeststatus: 'approved' },
    };

    wrapper = shallow(<SingleRequest {...props} match={match} />);
    expect(wrapper.find('.green').exists()).toBe(true);
  });
  it('Should display request label as red when status is dissapproved', () => {
    props = {
      aRequest: () => {},
      oneRequest:
        { requeststatus: 'disapproved' },
    };

    wrapper = shallow(<SingleRequest {...props} match={match} />);
    expect(wrapper.find('.red').exists()).toBe(true);
  });
  it('Should display request label as green when status is resolved', () => {
    props = {
      aRequest: () => {},
      oneRequest:
        { requeststatus: 'resolved' },
    };

    wrapper = shallow(<SingleRequest {...props} match={match} />);
    expect(wrapper.find('.green').exists()).toBe(true);
  });
});
