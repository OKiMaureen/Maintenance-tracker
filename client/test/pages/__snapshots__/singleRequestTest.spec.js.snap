// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleRequest /> stores a snapshot of the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleRequest
    aRequest={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    logout={[Function]}
    match={
      Object {
        "params": Object {
          "id": "1",
        },
      }
    }
    oneRequest={
      Object {
        "department": "technology",
        "description": "repair laptop",
        "equipment": "computer",
        "id": 1,
        "requeststatus": "pending",
        "serialnumber": 111111111,
        "title": "repair computer",
        "user_id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
          <Link
            replace={false}
            to="/createrequest"
          >
            Create Request
          </Link>
          <Link
            replace={false}
            to="/allrequests"
          >
            My request
          </Link>
          <button
            className="header-btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </Header>,
        <section
          className="allrequests"
        >
          <div
            className="detailsrequest-card"
          >
            <p>
              <label>
                Title:
              </label>
              repair computer
            </p>
            <p>
              <label>
                Department:
              </label>
              technology
            </p>
            <p>
              <label>
                Equipment: 
              </label>
              computer
            </p>
            <p>
              <label>
                Description: 
              </label>
              repair laptop
            </p>
            <div
              className="container"
            >
              <div
                className="status fixed"
              >
                <label
                  className="yellow"
                >
                  pending
                </label>
                
                
                
              </div>
              <div
                className="status fixed "
              >
                <p>
                  <Link
                    replace={false}
                    to="/editrequest"
                  >
                    Edit
                  </Link>
                </p>
              </div>
            </div>
          </div>
        </section>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>,
            <Link
              replace={false}
              to="/createrequest"
            >
              Create Request
            </Link>,
            <Link
              replace={false}
              to="/allrequests"
            >
              My request
            </Link>,
            <button
              className="header-btn"
              onClick={[Function]}
            >
              Logout
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Create Request",
              "replace": false,
              "to": "/createrequest",
            },
            "ref": null,
            "rendered": "Create Request",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My request",
              "replace": false,
              "to": "/allrequests",
            },
            "ref": null,
            "rendered": "My request",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "className": "header-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="detailsrequest-card"
          >
            <p>
              <label>
                Title:
              </label>
              repair computer
            </p>
            <p>
              <label>
                Department:
              </label>
              technology
            </p>
            <p>
              <label>
                Equipment: 
              </label>
              computer
            </p>
            <p>
              <label>
                Description: 
              </label>
              repair laptop
            </p>
            <div
              className="container"
            >
              <div
                className="status fixed"
              >
                <label
                  className="yellow"
                >
                  pending
                </label>
                
                
                
              </div>
              <div
                className="status fixed "
              >
                <p>
                  <Link
                    replace={false}
                    to="/editrequest"
                  >
                    Edit
                  </Link>
                </p>
              </div>
            </div>
          </div>,
          "className": "allrequests",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "5c5732",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <label>
                  Title:
                </label>
                repair computer
              </p>,
              <p>
                <label>
                  Department:
                </label>
                technology
              </p>,
              <p>
                <label>
                  Equipment: 
                </label>
                computer
              </p>,
              <p>
                <label>
                  Description: 
                </label>
                repair laptop
              </p>,
              <div
                className="container"
              >
                <div
                  className="status fixed"
                >
                  <label
                    className="yellow"
                  >
                    pending
                  </label>
                  
                  
                  
                </div>
                <div
                  className="status fixed "
                >
                  <p>
                    <Link
                      replace={false}
                      to="/editrequest"
                    >
                      Edit
                    </Link>
                  </p>
                </div>
              </div>,
            ],
            "className": "detailsrequest-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Title:
                  </label>,
                  "repair computer",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Title:",
                  },
                  "ref": null,
                  "rendered": "Title:",
                  "type": "label",
                },
                "repair computer",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Department:
                  </label>,
                  "technology",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Department:",
                  },
                  "ref": null,
                  "rendered": "Department:",
                  "type": "label",
                },
                "technology",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Equipment: 
                  </label>,
                  "computer",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Equipment: ",
                  },
                  "ref": null,
                  "rendered": "Equipment: ",
                  "type": "label",
                },
                "computer",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Description: 
                  </label>,
                  "repair laptop",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description: ",
                  },
                  "ref": null,
                  "rendered": "Description: ",
                  "type": "label",
                },
                "repair laptop",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="status fixed"
                  >
                    <label
                      className="yellow"
                    >
                      pending
                    </label>
                    
                    
                    
                  </div>,
                  <div
                    className="status fixed "
                  >
                    <p>
                      <Link
                        replace={false}
                        to="/editrequest"
                      >
                        Edit
                      </Link>
                    </p>
                  </div>,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label
                        className="yellow"
                      >
                        pending
                      </label>,
                      "",
                      "",
                      "",
                    ],
                    "className": "status fixed",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "pending",
                        "className": "yellow",
                      },
                      "ref": null,
                      "rendered": "pending",
                      "type": "label",
                    },
                    "",
                    "",
                    "",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      <Link
                        replace={false}
                        to="/editrequest"
                      >
                        Edit
                      </Link>
                    </p>,
                    "className": "status fixed ",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/editrequest"
                      >
                        Edit
                      </Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Edit",
                        "replace": false,
                        "to": "/editrequest",
                      },
                      "ref": null,
                      "rendered": "Edit",
                      "type": [Function],
                    },
                    "type": "p",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
            <Link
              replace={false}
              to="/createrequest"
            >
              Create Request
            </Link>
            <Link
              replace={false}
              to="/allrequests"
            >
              My request
            </Link>
            <button
              className="header-btn"
              onClick={[Function]}
            >
              Logout
            </button>
          </Header>,
          <section
            className="allrequests"
          >
            <div
              className="detailsrequest-card"
            >
              <p>
                <label>
                  Title:
                </label>
                repair computer
              </p>
              <p>
                <label>
                  Department:
                </label>
                technology
              </p>
              <p>
                <label>
                  Equipment: 
                </label>
                computer
              </p>
              <p>
                <label>
                  Description: 
                </label>
                repair laptop
              </p>
              <div
                className="container"
              >
                <div
                  className="status fixed"
                >
                  <label
                    className="yellow"
                  >
                    pending
                  </label>
                  
                  
                  
                </div>
                <div
                  className="status fixed "
                >
                  <p>
                    <Link
                      replace={false}
                      to="/editrequest"
                    >
                      Edit
                    </Link>
                  </p>
                </div>
              </div>
            </div>
          </section>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>,
              <Link
                replace={false}
                to="/createrequest"
              >
                Create Request
              </Link>,
              <Link
                replace={false}
                to="/allrequests"
              >
                My request
              </Link>,
              <button
                className="header-btn"
                onClick={[Function]}
              >
                Logout
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Create Request",
                "replace": false,
                "to": "/createrequest",
              },
              "ref": null,
              "rendered": "Create Request",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My request",
                "replace": false,
                "to": "/allrequests",
              },
              "ref": null,
              "rendered": "My request",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "className": "header-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Logout",
              "type": "button",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="detailsrequest-card"
            >
              <p>
                <label>
                  Title:
                </label>
                repair computer
              </p>
              <p>
                <label>
                  Department:
                </label>
                technology
              </p>
              <p>
                <label>
                  Equipment: 
                </label>
                computer
              </p>
              <p>
                <label>
                  Description: 
                </label>
                repair laptop
              </p>
              <div
                className="container"
              >
                <div
                  className="status fixed"
                >
                  <label
                    className="yellow"
                  >
                    pending
                  </label>
                  
                  
                  
                </div>
                <div
                  className="status fixed "
                >
                  <p>
                    <Link
                      replace={false}
                      to="/editrequest"
                    >
                      Edit
                    </Link>
                  </p>
                </div>
              </div>
            </div>,
            "className": "allrequests",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "5c5732",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  <label>
                    Title:
                  </label>
                  repair computer
                </p>,
                <p>
                  <label>
                    Department:
                  </label>
                  technology
                </p>,
                <p>
                  <label>
                    Equipment: 
                  </label>
                  computer
                </p>,
                <p>
                  <label>
                    Description: 
                  </label>
                  repair laptop
                </p>,
                <div
                  className="container"
                >
                  <div
                    className="status fixed"
                  >
                    <label
                      className="yellow"
                    >
                      pending
                    </label>
                    
                    
                    
                  </div>
                  <div
                    className="status fixed "
                  >
                    <p>
                      <Link
                        replace={false}
                        to="/editrequest"
                      >
                        Edit
                      </Link>
                    </p>
                  </div>
                </div>,
              ],
              "className": "detailsrequest-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Title:
                    </label>,
                    "repair computer",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Title:",
                    },
                    "ref": null,
                    "rendered": "Title:",
                    "type": "label",
                  },
                  "repair computer",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Department:
                    </label>,
                    "technology",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Department:",
                    },
                    "ref": null,
                    "rendered": "Department:",
                    "type": "label",
                  },
                  "technology",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Equipment: 
                    </label>,
                    "computer",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Equipment: ",
                    },
                    "ref": null,
                    "rendered": "Equipment: ",
                    "type": "label",
                  },
                  "computer",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Description: 
                    </label>,
                    "repair laptop",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Description: ",
                    },
                    "ref": null,
                    "rendered": "Description: ",
                    "type": "label",
                  },
                  "repair laptop",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="status fixed"
                    >
                      <label
                        className="yellow"
                      >
                        pending
                      </label>
                      
                      
                      
                    </div>,
                    <div
                      className="status fixed "
                    >
                      <p>
                        <Link
                          replace={false}
                          to="/editrequest"
                        >
                          Edit
                        </Link>
                      </p>
                    </div>,
                  ],
                  "className": "container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <label
                          className="yellow"
                        >
                          pending
                        </label>,
                        "",
                        "",
                        "",
                      ],
                      "className": "status fixed",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "pending",
                          "className": "yellow",
                        },
                        "ref": null,
                        "rendered": "pending",
                        "type": "label",
                      },
                      "",
                      "",
                      "",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        <Link
                          replace={false}
                          to="/editrequest"
                        >
                          Edit
                        </Link>
                      </p>,
                      "className": "status fixed ",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/editrequest"
                        >
                          Edit
                        </Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Edit",
                          "replace": false,
                          "to": "/editrequest",
                        },
                        "ref": null,
                        "rendered": "Edit",
                        "type": [Function],
                      },
                      "type": "p",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
